protocol/x/subaccounts/keeper/transfer.go

```diff
@@ -209,14 +209,77 @@ func (k Keeper) WithdrawFundsFromSubaccountToAccount(
 	)
 }
 
-// TransferFeesToFeeCollectorModule translates the assetId and quantums into a sdk.Coin,
-// and moves the funds from subaccounts module to the `fee_collector` module account by calling
-// bankKeeper.SendCoins(). Does not change any individual subaccount state.
-func (k Keeper) TransferFeesToFeeCollectorModule(
+// DistributeFees calculates the market mapper revenue share and fee collector share
+// based on the quantums and perpetual parameters, and transfers the fees to the
+// market mapper and fee collector.
+func (k Keeper) DistributeFees(
 	ctx sdk.Context,
 	assetId uint32,
 	quantums *big.Int,
 	perpetualId uint32,
+) error {
+	// get perpetual
+	perpetual, err := k.perpetualsKeeper.GetPerpetual(ctx, perpetualId)
+	if err != nil {
+		return err
+	}
+
+	collateralPoolAddr, err := k.GetCollateralPoolFromPerpetualId(ctx, perpetualId)
+	if err != nil {
+		return err
+	}
+
+	// calculate market mapper rev share
+	marketMapperShare := big.NewInt(0)
+	revShareAddr, revSharePpm, err := k.revShareKeeper.GetMarketMapperRevenueShareForMarket(
+		ctx,
+		perpetual.Params.MarketId,
+	)
+	if err == nil && revShareAddr != nil {
+		// marketMapperShare = quantums * revSharePpm / 1e6
+		marketMapperShare.Div(
+			new(big.Int).Mul(quantums, big.NewInt(int64(revSharePpm))),
+			big.NewInt(1e6),
+		)
+	}
+
+	// Remaining amount goes to the fee collector
+	feeCollectorShare := new(big.Int).Sub(quantums, marketMapperShare)
```

Comment:
non-blocking: What do you think about adding a check here that the sign doesn't flip between `quantums` and `feeCollectorShare` (revshare > 100% of fees)? If it does log an error and set revshare to 0?